twake/frontend/src/app/components/UserListManager/UserListManager.tsx

```diff
@@ -1,103 +1,256 @@
-import React, { FC, useState, useEffect } from 'react';
+import React, { useState, useEffect } from 'react';
+import Strings from 'app/services/utils/strings';
 import UsersService from 'services/user/user.js';
-import { UserType } from 'app/models/User';
-import { AutoComplete, Avatar, Col, Row, Typography } from 'antd';
+import Languages from 'services/languages/languages.js';
 import Workspaces from 'services/workspaces/workspaces.js';
+import UserOrMail from '../ui/UserOrMail';
 import TrashIcon from '@material-ui/icons/DeleteOutlined';
-import UserOrMail from 'components/ui/UserOrMail.js';
+import { Button, Col, Row, Typography } from 'antd';
+import Icon from '../Icon/Icon';
+import WorkspacesUsers from 'services/workspaces/workspaces_users.js';
 import AutoCompleteExtended from 'components/AutoCompleteExtended/AutoCompleteExtended';
+import { UserType } from 'app/models/User';
 import './UserListManager.scss';
 
-type Props = {
+type PropsType = {
   [key: string]: any;
 };
 
-const { Option } = AutoComplete;
-const { Text } = Typography;
+const UserListManager = (props: PropsType) => {
+  const [input, setInput] = useState<string>('');
+  const [editing, setEditing] = useState<boolean>(props.autoFocus ? props.autoFocus : false);
+  const [usersIds, setUsersIds] = useState<string[]>([...props.users]);
+  let savedUserProps: string;
+
+  useEffect(() => {
+    updateStateFromProps(props, true);
+  }, []);
+
+  const updateStateFromProps = (props: PropsType, force?: boolean) => {
+    let anti_duplicates: string[] = [];
 
-const UserListManager: FC<Props> = props => {
-  const [SelectedUsersList, _setSelectedUsersList] = useState<string[]>([]);
+    const user_ids = props.users
+      .map((item: any) => item.id || item)
+      .filter((item: string) => {
+        const here = anti_duplicates.indexOf(item) < 0;
```

Comment:
This duplicate block looks dirty. If you want to avoid dupes, use a `Set` instead of an `Array`. Or use `lodash` to remove duplicates, code will be cleaner